k9mail/src/main/java/com/fsck/k9/ui/messageview/MessageTopView.java

```diff
@@ -282,8 +285,11 @@ private void hideShowPicturesButton() {
         showPicturesButton.setVisibility(View.GONE);
     }
 
-    private boolean shouldAutomaticallyLoadPictures(ShowPictures showPicturesSetting, Message message) {
-        return showPicturesSetting == ShowPictures.ALWAYS || shouldShowPicturesFromSender(showPicturesSetting, message);
+    private boolean shouldAutomaticallyLoadPictures(ShowPictures showPicturesSetting, boolean onlyShowPicturesViaWiFi,
+            Message message) {
+        return (showPicturesSetting == ShowPictures.ALWAYS ||
+                shouldShowPicturesFromSender(showPicturesSetting, message)) &&
```

Comment:
I think so.. 
```
showPicturesSetting == ShowPictures.ALWAYS || shouldShowPicturesFromSender(showPicturesSetting, message)
```
indicates whether the user want images from this sender to show, 
```
shouldShowPicturesInCurrentNetwork(onlyShowPicturesViaWiFi)
```
indicates whether the user want images to show on current network. Images should be shown when both are met.